python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,262 @@
+
+    def _convert_hive_into_iceberg(self, _: HiveTable) -> Table:
+        # Requires reading the manifest, will implement this in another PR
+        # Also nice to have the REST catalog in first
+        return Table()
+
+    def create_table(
+        self,
+        identifier: Union[str, Identifier],
+        schema: Schema,
+        location: Optional[str] = None,
+        partition_spec: Optional[PartitionSpec] = None,
+        properties: Optional[Properties] = None,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        current_time_millis = int(time())
+        tbl = HiveTable(
+            dbName=database_name,
+            tableName=table_name,
+            owner=getuser(),
+            createTime=current_time_millis // 1000,
+            lastAccessTime=current_time_millis // 1000,
+            sd=self._storage_descriptor(schema, location),
+            tableType="EXTERNAL_TABLE",
+            parameters={"EXTERNAL": "TRUE"},
+        )
+        try:
+            with self._client as open_client:
+                open_client.create_table(tbl)
+                hive_table = open_client.get_table(dbname=database_name, tbl_name=table_name)
+        except AlreadyExistsException as e:
+            raise AlreadyExistsError(f"Table {database_name}.{table_name} already exists") from e
+        return self._convert_hive_into_iceberg(hive_table)
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            with self._client as open_client:
+                hive_table = open_client.get_table(dbname=database_name, tbl_name=table_name)
+        except NoSuchObjectException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} not found") from e
+
+        return self._convert_hive_into_iceberg(hive_table)
+
+    def _drop_table(self, identifier: Union[str, Identifier], delete_data: bool = False) -> None:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            with self._client as open_client:
+                open_client.drop_table(dbname=database_name, name=table_name, deleteData=delete_data)
+        except NoSuchObjectException as e:
+            # When the namespace doesn't exists, it throws the same error
+            raise NoSuchTableError(f"Table {table_name} cannot be found") from e
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        self._drop_table(identifier)
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        self._drop_table(identifier, True)
+
+    def rename_table(self, from_identifier: Union[str, Identifier], to_identifier: Union[str, Identifier]) -> Table:
+        from_database_name, from_table_name = self.identifier_to_tuple(from_identifier)
+        to_database_name, to_table_name = self.identifier_to_tuple(to_identifier)
```

Comment:
Hive only supports one namespace level, so we can throw `NoSuchTableException` if the `from` identifier has more than one namespace element. Similarly, this should fail with a reasonable error (ValueError probably) if the destination namespace is invalid.

Reply:
Good point, I've added checks to catch this.